py/makeqstrdata.py

```diff
@@ -222,6 +223,71 @@
     "zip",
 ]
 
+# Additional QSTRs that must have index <255 because they are stored in
+# `mp_binary_op_method_name` and `mp_unary_op_method_name` (see py/objtype.c).
+# These are not part of the .mpy compatibility list, but we place them in the
+# fixed unsorted pool (i.e. QDEF0) to ensure their indices are small.
+operator_qstr_list = {
```

Comment:
I'm guessing you thought of this and it's not worth it, but I noticed this list is about 26% of the QDEF0 pool (57/(165+57)). Given the `qstr.c` lookup code already iterates an array of `pool`s, is it worth adding another sorted pool for these, placed after the static pool? Or is 57 lookups so small that it's better to do it this way?

Reply:
Yep, we did consider this, and actually the static list is _almost_ sorted anyway. So we could turn this into three sorted pools (and then special-case the first two entries). However, each pool has a (small) size overhead, but also a performance overhead for turning a qstr ID into a hash or a length (which go via `find_qstr`). (Getting the actual bytes of a qstr is actually not something that's really performance critical, but hash and length are).

So i was going to leave this (and further investigation) for another PR with its own performance comparison.

(Sorry I should have mentioned that we'd already thought about this when we spoke about this PR!)